llvm/lib/CodeGen/ExpandVectorPredication.cpp

```diff
@@ -465,6 +465,21 @@ CachingVPExpander::expandPredicationInMemoryIntrinsic(IRBuilder<> &Builder,
         VPI.getName());
     break;
   }
+  case Intrinsic::vp_load_ff: {
+    // Mask out all but the first lane.
+    Value *NewMask = ConstantInt::getFalse(MaskParam->getType());
+    NewMask = Builder.CreateInsertElement(
+        NewMask, ConstantInt::getTrue(MaskParam->getType()->getScalarType()),
+        (uint64_t)0);
+    NewMask = Builder.CreateAnd(MaskParam, NewMask);
+    Value *MaskedLoad = Builder.CreateMaskedLoad(
+        VPI.getType()->subtypes()[0], PtrParam, AlignOpt.valueOrOne(), NewMask);
+    Value *EVLResult = Builder.getInt32(1);
```

Comment:
Should the returned scalar be `umin(EVL, 1)` in case EVL is 0? The LangRef seems to suggest that it should be but isn't strict about it:

> The second result is usually less than evl when an exception would occur for reading that lane

Reply:
The original EVL has been moved to the mask by this point. The EVL this code sees will be VLMax so its always more than 1.

Reply:
I think that would introduce the forward progress problem Philip mentioned. Maybe we need to do this expansion before the EVL rewrite?